lib_eio_linux/low_level.ml

```diff
@@ -366,6 +375,33 @@ let mkdir_beneath ~perm dir path =
   try eio_mkdirat parent leaf perm
   with Unix.Unix_error (code, name, arg) -> raise @@ Err.wrap_fs code name arg
 
+let float_of_time (s, ns) = Int64.to_float s +. (float ns /. 1e9)
```

Comment:
OCaml itself makes sure the integer part isn't changed by rounding (https://github.com/ocaml/ocaml/issues/9490). We should probably do that too (here and in eio_posix).